src/azure-cli-core/azure/cli/core/__init__.py

```diff
@@ -30,6 +30,9 @@
 EXCLUDED_PARAMS = ['self', 'raw', 'polling', 'custom_headers', 'operation_config',
                    'content_version', 'kwargs', 'client', 'no_wait']
 EVENT_FAILED_EXTENSION_LOAD = 'MainLoader.OnFailedExtensionLoad'
+_COMMAND_INDEX = 'commandIndex'
+# Extensions that will always be loaded if installed. These extensions don't expose commands but hook into CLI core.
+ALWAYS_LOADED_EXTENSION_MODNAMES = ['azext_ai_examples', 'azext_ai_did_you_mean_this']
```

Comment:
@mirdaki, @christopher-o-toole, I have allowed [`azext_ai_examples`](https://github.com/Azure/azure-cli-extensions/tree/master/src/ai-examples) (https://github.com/Azure/azure-cli-extensions/pull/1345) and `azext_ai_did_you_mean_this` (https://github.com/Azure/azure-cli-extensions/pull/1536) to be always loaded. Please member to change this const list in CLI core if you are going to change the namespace in the future.

Reply:
what if the extensions not installed ?

Reply:
This is only a filter applied to installed extensions. If they are not installed, they will simply be ignored. https://github.com/Azure/azure-cli/blob/3632dc0fa87a4c9c928f204cfb61d448c304a208/src/azure-cli-core/azure/cli/core/__init__.py#L272-L279